### PR TITLE
Correct relative mouse input handling

### PR DESCRIPTION
**Pull request by necessarily-equal**
Closes https://github.com/DaemonEngine/Daemon/issues/600. Hopefully for good this time.

### CODE REVIEW

**Comment by necessarily-equal** [comment]
I've tested this to work on SDL2.0.20, SDL2.0.22 and SDL main branch (as of https://github.com/libsdl-org/SDL/commit/3807e95029d77a480e80adddd7c0f21abc29baa9).

Testing with old SDL versions is left as an exercise for the reader ;-)

**Comment by slipher** [comment]
This patch breaks Mac by making it impossible to move the mouse outside the window. This is true with both SDL 2.0.12 and latest SDL.

File: src/engine/sys/sdl_input.cpp [hunk 1]
```diff
@@ -427,7 +427,7 @@ void IN_SetMouseMode(MouseMode newMode)
 		}
 
 		SDL_ShowCursor( newMode == MouseMode::SystemCursor ? SDL_ENABLE : SDL_DISABLE );
-		auto grab = newMode != MouseMode::Deltas || in_nograb->integer ? SDL_FALSE : SDL_TRUE;
+		auto grab = in_nograb->integer ? SDL_FALSE : SDL_TRUE;
```
Comment (necessarily-equal) [inline src/engine/sys/sdl_input.cpp:430]: @slipher do it work for you if I remove this change?

Comment (slipher) [reply src/engine/sys/sdl_input.cpp:430]: Yes, the bug goes away upon reverting that part.

Comment (necessarily-equal) [reply src/engine/sys/sdl_input.cpp]: Great, reverting that part then.

**Comment by necessarily-equal** [comment]
@bmorel since you seem to use a minimalist X11 setup, you may want to try to see if this PR don't introduce some regression on your setup.

```C
// work around X window managers and edge-based workspace flipping
// - without this, we get LeaveNotify, no mouse button events, EnterNotify;
//   we would likely miss some button presses and releases.
IN_CenterMouse();
```

I'd say you don't really have to try it either, you can also complain afterwards ;)

**Comment by slipher** [comment]
The workspace flipping thing described in the comment seems to be an xfce behavior. https://askubuntu.com/questions/891859/xubuntu-snap-to-window-switching-workspaces

**Comment by necessarily-equal** [comment]
Tested working by @IngarKCT. It also works for me and it fix a problem that totally breaks the game, so I'm merging this.

I'll have a look at the XFCE situation a bit later, to check if it is still a problem.

**Comment by necessarily-equal** [comment]
I've checked XFCE and I've ticked that thing in the parameters. I 
couldn't reproduce the bug with the fixed daemon.

**Comment by necessarily-equal** [comment]
@Viech it would be nice if you could backport this to the AUR

This patch should* apply cleanly [sdl.patch.txt](https://github.com/DaemonEngine/Daemon/files/8935634/sdl.patch.txt) (although it was hand-edited, and is untested)

**Pull request merged**